base-system/tools/installer/install.sh

```diff
@@ -102,11 +98,28 @@ fi
 
 ## For each mountpoint that the device is using, kill and unmount
 print_step "[4/11] Unmounting selected device partitions"
-for MNT_PNT in $(lsblk --output PATH,MOUNTPOINT | grep -E "${TARGET}[1-9]+" | awk '{ print $2 }'); do
-	echo "Cleaning $MNT_PNT"
-	fuser -k -m "$MNT_PNT" || true
-	umount "$MNT_PNT"
-done
+# Count every target device's mounted partitions
+PARTITION_COUNT=$(mount | grep "$TARGET" | awk '{print $3}' | wc -l)
+
+# If mounted partitions exists, unmount them.
+if [[ "${PARTITION_COUNT}" -gt 0 ]]; then
+  echo "Unmounting $PARTITION_COUNT partition(s)"
+  PARTITIONS="$(mount | grep "$TARGET" | awk '{print $3}')"
+  for PARTIITON in ${PARTITIONS}; do
+    echo "Unmounting '$PARTIITON'"
+    # Kill proceses interacting with the partition, if any
+    fuser -k -m "$PARTIITON" || true
+    # Unmount the partition
+    umount "$PARTIITON" || exit 1 #Error unmounting
+  done
+  echo "Partition(s) unmounted correctly"
+else
+  echo "No partitions to unmount"
+fi
+
+# Wipes device's filesystem
+wipefs -a "$TARGET" || exit 1
```

Comment:
> Does this _wipefs_ is something standard? Also, is there's some way in which this might take longer that expected, in which case might be better to print an aditional step here.

Ok, I've just investigated about *wipefs* and seems to be very clever solution. Just dumping all the partition tables and signatures instead of zeroing. Amazing. Thanks for this!